src/pairinteraction/_wrapped/system/system.py

```diff
@@ -83,7 +84,7 @@ def diagonalize(
     ) -> "Self":
         """Diagonalize the Hamiltonian and update the basis to the eigenbasis.
 
-        This method computes the eigenvalues and eigenvectors of the Hamiltonian
+        This method computes the eigenenergies and eigenbasis of the Hamiltonian
```

Comment:
I would rather write "This method diagonalizes the Hamiltonian of the system and updates the basis of the system accordingly."